app/donut.py

```diff
@@ -1,3 +1,27 @@
-async def donut_handler(args):
-    print('This is where we would configure things to properly handle a .donut file request.')
-    return '', ''
+import donut
+import os
+import shutil
+
+
+async def donut_handler(services, args):
+    _, file_name = await services.get('file_svc').find_file_path(args.get('file'), location='payloads')
+    exe_path, donut_ext = _stage_compatible_executable(file_name)
+    shellcode = donut.create(file=exe_path)
+    _write_shellcode_to_file(shellcode, file_name)
+    os.remove(exe_path)
+    return donut_ext, donut_ext
```

Comment:
I don't understand this line. why return the same thing twice?

Reply:
it's returning to the `file_svc` display_name and payload_name.  Idea being we still want to allow for obfuscation in the future

Reply:
I'll change to just returning the payload_name

Reply:
Looking at it... I'd prefer to keep the flexibility:
```Python
payload, display_name = await self._operate_extension(payload, headers)
```